scripts/createDB.sql

```diff
@@ -0,0 +1,335 @@
+-- Create a local db user in order for the backend to be able to access the data.
+drop user if exists 'admin'@'localhost';
+create user 'admin'@'localhost' identified by 'admin';
+grant all privileges on * . * to 'admin'@'localhost';
+ALTER USER 'admin'@'localhost' IDENTIFIED WITH mysql_native_password BY 'admin';
+flush PRIVILEGES;
+
+drop schema if exists `soen_390_db`;
+
+create schema `soen_390_db`;
+
+USE `soen_390_db`;
+
+CREATE TABLE `soen_390_db`.`user` (
+  `userID` INT NOT NULL AUTO_INCREMENT,
+  `name` VARCHAR(45) NOT NULL,
+  `role` VARCHAR(45) NOT NULL,
+  `email` VARCHAR(45) NOT NULL,
+  `password` CHAR(60) NOT NULL,
+  UNIQUE INDEX `email_UNIQUE` (`email` ASC) VISIBLE,
+  PRIMARY KEY (`userID`))
+
+  collate = utf8mb4_unicode_ci;
+  
+CREATE TABLE `soen_390_db`.`customer_order` (
+  `customer_order_id` INT NOT NULL AUTO_INCREMENT,
+  `status` VARCHAR(45) NOT NULL,
+  `date_time_added` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `date_time_finished` DATETIME,
+  `price` DECIMAL(10,2) NOT NULL,
+  PRIMARY KEY (`customer_order_id`))
+  
+  collate = utf8mb4_unicode_ci;
+  
+CREATE TABLE `soen_390_db`.`production_order` (
+  `production_order_id` INT NOT NULL AUTO_INCREMENT,
+  `status` VARCHAR(45) NOT NULL,
+  `date_time_added` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `date_time_finished` DATETIME,
+  `price` DECIMAL(10,2) NOT NULL,
+  PRIMARY KEY (`production_order_id`))
+  
+  collate = utf8mb4_unicode_ci;
+
+CREATE TABLE `soen_390_db`.`raw_material` (
+  `raw_material_id` INT NOT NULL AUTO_INCREMENT,
+  `raw_material_name` VARCHAR(45) NOT NULL,
+  `quantity` INT NOT NULL,
+  `price` DECIMAL(10,2) NOT NULL,
+  `vendor` VARCHAR(45),
+  `date_time_added` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY (`raw_material_id`),
+  UNIQUE INDEX `raw_material_name_UNIQUE` (`raw_material_name` ASC) VISIBLE)
+  
+  collate = utf8mb4_unicode_ci;
+    
+CREATE TABLE `soen_390_db`.`semi_finished_good` (
+  `semi_finished_good_id` INT NOT NULL AUTO_INCREMENT,
+  `semi_finished_good_name` VARCHAR(45) NOT NULL,
+  `quantity` INT NOT NULL,
+  `price` DECIMAL(10,2) NOT NULL,
+  `vendor` VARCHAR(45),
```

Comment:
Remove vendor? because we build this item?

Reply:
I added vendor because while we can build a semi finished good, we can also buy one. For example, we said that we consider paint as a semi finished good and we will buy paint (we won't make it). That's why I suggest we keep this field.